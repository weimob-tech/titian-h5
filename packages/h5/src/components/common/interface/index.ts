export type OmitType<T, P> = Pick<T, { [key in keyof T]: T[key] extends P ? never : key }[keyof T]>;

export type Key = string | number | symbol;

export type IAnyObject = { [key: Key]: unknown };

export type IAnyMap = Map<Key, unknown>;

export type DistanceUnit = 'px' | 'em' | 'rem' | 'vw' | 'vh' | '%';

export type IPosition = {
  bottom?: `${number}${DistanceUnit}`;
  top?: `${number}${DistanceUnit}`;
  left?: `${number}${DistanceUnit}`;
  right?: `${number}${DistanceUnit}`;
};
