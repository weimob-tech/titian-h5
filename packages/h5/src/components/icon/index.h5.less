@import '../common/static/utils.less';
@import '../common/static/var.less';

@fontfamily:~ '@{project}-icon';
@fontfamilylovely:~ '@{project}-icon-lovely';
@fontfamilypopular:~ '@{project}-icon-popular';

@font-face {
  font-family: @fontfamily; /* Project id 2512151 */
  src: url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.woff2?t=1670393250377') format('woff2'),
    url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.woff?t=1670393250377') format('woff'),
    url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.ttf?t=1670393250377') format('truetype');
}

@font-face {
  font-family: @fontfamilylovely; /* Project id 2914379 */
  src: url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.woff2?t=1661930836565') format('woff2'),
    url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.woff?t=1661930836565') format('woff'),
    url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.ttf?t=1661930836565') format('truetype');
}

@font-face {
  font-family: @fontfamilypopular; /* Project id 2914381 */
  src: url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.woff2?t=1661930832349') format('woff2'),
    url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.woff?t=1661930832349') format('woff'),
    url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.ttf?t=1661930832349') format('truetype');
}

@icon-family: var(--icon-family, @fontfamily);

@protected-icon-family: var(--protected-icon-family, @icon-family);

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(1turn);
  }
}

@icon:~ '@{project}-icon';

.@{icon}-base {
  @protected-icon-color: var(--protected-icon-color, @icon-color);
  @protected-icon-font-size: var(--protected-icon-font-size, @icon-font-size);

  color: @protected-icon-color;
  font-size: @protected-icon-font-size;

  &&-spin {
    animation: spin 1s linear infinite;
  }
}

.@{icon} {
  @protected-icon-display: var(--protected-icon-display, @icon-display);
  @protected-icon-line-height: var(--protected-icon-line-height, @icon-line-height);

  display: @protected-icon-display;
  position: relative;

  font-family: @protected-icon-family;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  line-height: @protected-icon-line-height;

  &&-right-custom::before {
    content: '\e61c';
  }

  &&-share::before {
    content: '\e6cf';
  }

  &&-notice::before {
    content: '\e6cc';
  }

  &&-addressright::before {
    content: '\e617';
  }

  &&-addressreceiving::before {
    content: '\e618';
  }

  &&-intracity::before {
    content: '\e6c9';
  }

  &&-delivery::before {
    content: '\e6ca';
  }

  &&-checkbox-hollow::before {
    content: '\e604';
  }

  &&-douyin::before {
    content: '\e66b';
  }

  &&-alipay::before {
    content: '\e66c';
  }

  &&-kuaishou::before {
    content: '\e66d';
  }

  &&-xiaohongshu::before {
    content: '\e66e';
  }

  &&-baidu::before {
    content: '\e66f';
  }

  &&-checkbox::before {
    content: '\e665';
  }

  &&-location::before {
    content: '\e6b8';
  }

  &&-selected::before {
    content: '\e602';
  }

  &&-phone::before {
    content: '\e69c';
  }

  &&-user-account-setting::before {
    content: '\e69a';
  }

  &&-colon::before {
    content: '\e649';
  }

  &&-error::before {
    content: '\e699';
  }

  &&-right::before {
    content: '\e698';
  }

  &&-list::before {
    content: '\e697';
  }

  &&-view::before {
    content: '\e696';
  }

  &&-recommend::before {
    content: '\e695';
  }

  &&-speaker::before {
    content: '\e694';
  }

  &&-profile::before {
    content: '\e693';
  }

  &&-home::before {
    content: '\e692';
  }

  &&-cart::before {
    content: '\e691';
  }

  &&-connect::before {
    content: '\e690';
  }

  &&-add-plain::before {
    content: '\e68e';
  }

  &&-arrange::before {
    content: '\e68d';
  }

  &&-category::before {
    content: '\e68c';
  }

  &&-phone-plain::before {
    content: '\e68a';
  }

  &&-warning::before {
    content: '\e689';
  }

  &&-share-hwq::before {
    content: '\e688';
  }

  &&-play-plain::before {
    content: '\e687';
  }

  &&-vip::before {
    content: '\e685';
  }

  &&-wechat::before {
    content: '\e642';
  }

  &&-qq::before {
    content: '\e643';
  }

  &&-weibo::before {
    content: '\e644';
  }

  &&-share-wechat-moments::before {
    content: '\e645';
  }

  &&-save::before {
    content: '\e63c';
  }

  &&-qrcode::before {
    content: '\e63d';
  }

  &&-goods::before {
    content: '\e63f';
  }

  &&-link::before {
    content: '\e640';
  }

  &&-picture::before {
    content: '\e641';
  }

  &&-clock-alarm::before {
    content: '\e63b';
  }

  &&-tabbar-category-highlight::before {
    content: '\e631';
  }

  &&-tabbar-cart-highlight::before {
    content: '\e632';
  }

  &&-tabbar-mine-highlight::before {
    content: '\e633';
  }

  &&-tabbar-vip-highlight::before {
    content: '\e634';
  }

  &&-tabbar-home-highlight::before {
    content: '\e635';
  }

  &&-tabbar-cart::before {
    content: '\e636';
  }

  &&-tabbar-home::before {
    content: '\e637';
  }

  &&-tabbar-vip::before {
    content: '\e638';
  }

  &&-tabbar-mine::before {
    content: '\e639';
  }

  &&-tabbar-category::before {
    content: '\e63a';
  }

  &&-search::before {
    content: '\e630';
  }

  &&-default-pic::before {
    content: '\e62f';
  }

  &&-rate-star-highlight::before {
    content: '\e62d';
  }

  &&-rate-star::before {
    content: '\e62e';
  }

  &&-go-to-top::before {
    content: '\e62c';
  }

  &&-to-receive::before {
    content: '\e627';
  }

  &&-mine-to-pay::before {
    content: '\e628';
  }

  &&-to-refund::before {
    content: '\e629';
  }

  &&-to-comment::before {
    content: '\e62a';
  }

  &&-to-deliver::before {
    content: '\e62b';
  }

  &&-camera-point::before {
    content: '\e620';
  }

  &&-clear::before {
    content: '\e611';
  }

  &&-question::before {
    content: '\e61b';
  }

  &&-store::before {
    content: '\e61a';
  }

  &&-info::before {
    content: '\e619';
  }

  &&-filter::before {
    content: '\e60a';
  }

  &&-stepper-minus::before {
    content: '\e603';
  }

  &&-checked::before {
    content: '\e605';
  }

  &&-minus::before {
    content: '\e606';
  }

  &&-stepper-plus::before {
    content: '\e607';
  }

  &&-positioning::before {
    content: '\e609';
  }

  &&-plus::before {
    content: '\e60b';
  }

  &&-sort-inactive::before {
    content: '\e60c';
  }

  &&-arrow-down::before {
    content: '\e60d';
  }

  &&-required::before {
    content: '\e60e';
  }

  &&-arrow-up::before {
    content: '\e60f';
  }

  &&-arrow-right::before {
    content: '\e610';
  }

  &&-address::before {
    content: '\e612';
  }

  &&-scan::before {
    content: '\e613';
  }

  &&-close::before {
    content: '\e614';
  }

  &&-delete::before {
    content: '\e615';
  }

  &&-nav-back::before {
    content: '\e616';
  }
}
