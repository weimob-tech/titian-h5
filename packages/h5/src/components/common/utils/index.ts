import { getTitianMode } from '../../../global/global';
import { IAnyObject, OmitType } from '../interface/index';

export * from './number';

export const objectToString = Object.prototype.toString;

export const toTypeString = (val: unknown): string => objectToString.call(val);

export type Key = string | number | symbol;

export const isString = (obj: unknown): obj is string => toTypeString(obj) === '[object String]';

export const isPlainObject = (obj: unknown): obj is { [key: Key]: unknown } => toTypeString(obj) === '[object Object]';

export const isPlainArray = (obj: unknown): obj is unknown[] => toTypeString(obj) === '[object Array]';

export const isPlainRegExp = (obj: unknown): obj is RegExp => toTypeString(obj) === '[object RegExp]';

export const isBoolean = (obj: unknown): obj is boolean => toTypeString(obj) === '[object Boolean]';

export const isFunction = (obj: unknown): obj is (...agrn: unknown[]) => unknown => typeof obj === 'function';

export const isArrayEqual = (arr1: unknown[], arr2: unknown[]): boolean => {
  if (arr1.length !== arr2.length) {
    return false;
  }

  return arr1.every((item, index) => item === arr2[index]);
};

export function debounce(fn: (...agrn: unknown[]) => unknown, ms = 300) {
  let timeoutId: ReturnType<typeof setTimeout>;
  return function debounceFn(this: unknown, ...args: unknown[]) {
    clearTimeout(timeoutId);
    timeoutId = setTimeout(() => fn.apply(this, args), ms);
  };
}

export function throttle(fn: (...agrn: unknown[]) => unknown, ms = 300) {
  let inThrottle: boolean;
  let lastFn: ReturnType<typeof setTimeout>;
  let lastTime: number;
  return function r(this: unknown, ...args: unknown[]) {
    if (!inThrottle) {
      fn.apply(this, args);
      lastTime = Date.now();
      inThrottle = true;
    } else {
      clearTimeout(lastFn);
      lastFn = setTimeout(() => {
        if (Date.now() - lastTime >= ms) {
          fn.apply(this, args);
          lastTime = Date.now();
        }
      }, Math.max(ms - (Date.now() - lastTime), 0));
    }
  };
}

export const randomString = () => Math.random().toString(36).slice(2);

export function filterInvalidData<T extends IAnyObject>(obj: T): OmitType<T, null | undefined> {
  const params = { ...obj };
  Object.keys(obj).forEach(key => {
    const value = obj[key];
    if (typeof value === 'undefined' || value === null) {
      delete params[key];
    }
  });
  return params;
}

export const DISABLED = 'disabled';

export function isDisabled(params: unknown): boolean {
  if (typeof params !== 'object' || !params) {
    return false;
  }
  return (params as IAnyObject)[DISABLED] as boolean;
}

export function range(num: number, min: number, max: number) {
  return Math.min(Math.max(num, min), max);
}

export function padZero(val: string | number, number = 2) {
  return `${val}`.padStart(number, '0');
}

export function isValidDate(d: unknown): d is Date {
  return d !== undefined && d !== null && !Number.isNaN(new Date(d as Date).getTime());
}

export function getDate(): Date;
export function getDate(date: string | number | Date): Date;
export function getDate(year: number, month: number): Date;
export function getDate(year: number, month: number, day: number): Date;
export function getDate(date?: string | number | Date, month?: number, day?: number) {
  let cur: Date;
  if (typeof date === 'undefined') {
    return new Date();
  }
  if (typeof date === 'number' && typeof month === 'number' && typeof day === 'number') {
    cur = new Date(date, month, day);
  } else if (typeof date === 'number' && typeof month === 'number') {
    cur = new Date(date, month);
  } else {
    cur = new Date(date);
  }
  if (!cur) {
    throw new Error('时间格式错误');
  }
  if (cur.toString() === 'Invalid Date') {
    throw new Error('时间格式错误');
  }
  return cur;
}

export function isUndefined(val: unknown) {
  return toTypeString(val) === '[object Undefined]';
}

// eslint-disable-next-line @titian-design/no-more-args
export function easeInOutCubic(t: number, b: number, c: number, d: number) {
  const cc = c - b;
  t /= d / 2;
  if (t < 1) {
    return (cc / 2) * t * t * t + b;
  }
  return (cc / 2) * ((t -= 2) * t * t + 2) + b;
}

// eslint-disable-next-line @titian-design/no-more-args
export function addEventListener<K extends Element>(
  target: K,
  eventType: string,
  cb: (arg: any) => any,
  options?: boolean | AddEventListenerOptions,
): () => void {
  if (target.addEventListener) {
    target.addEventListener(eventType, cb, options);
    document.addEventListener(eventType, cb, options);
  }

  return () => {
    if (target.removeEventListener) {
      target.removeEventListener(eventType, cb, options);
    }
  };
}

export const attrStyleToString = (attr: unknown) => {
  if (!attr) {
    return '';
  }
  if (typeof attr === 'string') {
    return attr;
  }
  if (Array.isArray(attr)) {
    return attr.join(';');
  }
  if (Object.keys(attr as { [key: string]: unknown }).length > 0) {
    return Object.keys(attr as { [key: string]: unknown })
      .reduce((array, key) => {
        array.push(`${key}:${(attr as { [key: string]: unknown })[key]}`);
        return array;
      }, [] as string[])
      .join(';');
  }

  return '';
};

export const stringToAttrStyle = (attrStr: unknown): { [key: string]: string } => {
  if (!attrStr) {
    return {};
  }
  if (isPlainObject(attrStr)) {
    return attrStr as { [key: string]: string };
  }
  if (!isString(attrStr)) {
    return {};
  }
  try {
    attrStr = JSON.parse(attrStr);
    // eslint-disable-next-line no-empty
  } catch (e) {}
  if (isPlainObject(attrStr)) {
    return attrStr as { [key: string]: string };
  }
  if (!isString(attrStr)) {
    return {};
  }
  return attrStr
    .trim()
    .split(';')
    .filter(Boolean)
    .reduce((pre, cur) => {
      const [key, value] = cur.split(':');
      const nkey = key.trim();
      pre[nkey] = value;
      return pre;
    }, {} as { [key: string]: string });
};

export const getResponsivePixel = (px: number) => {
  const mode = getTitianMode();
  if (mode === 'pc') {
    return px / 2;
  }
  return (document.documentElement.clientWidth / 750) * px;
};

export function getStyle<T extends HTMLElement>(element: T, attrName: keyof CSSStyleDeclaration) {
  let ret = element.style[attrName];
  if (ret) {
    return ret;
  }

  const getComputedStyle = document?.defaultView?.getComputedStyle;

  if (getComputedStyle) {
    ret = getComputedStyle(element, null)[attrName];
  }
  return ret;
}

export function setStyle(
  dom: HTMLElement,
  style: Partial<{ [key in keyof CSSStyleDeclaration]: CSSStyleDeclaration[key] | null }>,
) {
  (Object.keys(style) as (keyof CSSStyleDeclaration)[]).forEach(key => {
    if (style[key] === null) {
      dom.style.removeProperty(key as string);
    } else {
      dom.style.setProperty(key as string, style[key] as string);
    }
  });
}
export function hasClass(target: Element, className: string) {
  return target.className.indexOf(className) > -1;
}
export function addClass(target: Element, className: string) {
  return target.classList.add(className);
}
export function removeClass(target: Element, className: string) {
  return target.classList.remove(className);
}
export function toggleClass(target: Element, className: string) {
  return target.classList.toggle(className);
}
export function isCustomEvent(event: CustomEvent | Event): event is CustomEvent {
  return !event.isTrusted;
}

//  end 向后 无法查找 则  start end 之间
export function nextAvailable<T>(params: {
  start: number;
  end: number;
  list: T[];
  fn?: (list: T[], num: number) => boolean;
}) {
  const { start, end, list, fn } = params;
  let num = end;
  while (num < list.length) {
    num += 1;
    if (typeof list[num] !== undefined) {
      if (typeof fn === 'function' && fn(list, num)) {
        return num;
      }
    }
  }
  num = end;
  while (num > start) {
    num -= 1;
    if (typeof list[num] !== undefined) {
      if (typeof fn === 'function' && fn(list, num)) {
        return num;
      }
    }
  }
  return -1;
}

//  end 向前 无法查找 则 start end 之间
export function prevAvailable<T>(params: {
  start: number;
  end: number;
  list: T[];
  fn?: (list: T[], num: number) => boolean;
}) {
  const { start, list, end, fn } = params;
  let num = end;
  // 节点向前
  while (num > 0) {
    num -= 1;
    if (typeof list[num] !== undefined) {
      if (typeof fn === 'function' && fn(list, num)) {
        return num;
      }
    }
  }
  num = end;
  while (num < start) {
    num += 1;
    if (typeof list[num] !== undefined) {
      if (typeof fn === 'function' && fn(list, num)) {
        return num;
      }
    }
  }
  return -1;
}

export function addShadowRootStyle() {
  if (!this.extCss || !this.host || !this.host.shadowRoot) return;
  const style = this.host.shadowRoot.querySelector('style') || document.createElement('style');
  style.innerHTML = this.extCss;
  this.host.shadowRoot.appendChild(style);
}

export function pxToVW(str) {
  const mode = getTitianMode();

  if (mode === 'pc') return str;

  const res = (Number(str.replace('px', '')) * 100) / 750;
  function toFixedFn(value, n) {
    return Math.round(value * 10 ** n) / 10 ** n;
  }
  return `${toFixedFn(res, 5)}vw`;
}

export function validateEvent(event, shadowRoot) {
  const root = event.target.getRootNode();
  if (root === shadowRoot) {
    return true;
  }
  const slot = event.path.find(element => element.tagName === 'SLOT');
  const [node] = slot.assignedNodes();
  if (node && node.hasAttribute('slot')) {
    const stop = node.getAttribute('stopPropagation') || node.getAttribute('stop-propagation');
    if (typeof stop === 'boolean' && stop) {
      return false;
    }
    if (typeof stop === 'string' && (stop === 'true' || stop === '')) {
      return false;
    }
  }
  return true;
}

export function checkSlotIsUsed(host) {
  const slot = host.shadowRoot?.querySelector('slot')?.assignedNodes({ flatten: true });
  if (!slot || slot.length === 0) return false;
  if (slot && slot.length === 1) {
    if (slot[0].nodeName === '#text' && slot[0].textContent === '') {
      return false;
    }
  }
  return true;
}
