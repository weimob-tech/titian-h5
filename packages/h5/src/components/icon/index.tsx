import { Component, h, Host, Prop } from '@stencil/core';
import { JSXBase } from '@stencil/core/internal';
import { store } from '../common/basic/store';
import { stringToAttrStyle } from '../common/utils';
import { join, handle, PREFIX } from '../common/utils/namespace';
import addUnit from '../common/utils/suffix';

function completedStyles(data) {
  const style = [];
  if (data.extStyle) {
    if (typeof data.extStyle === 'string') {
      style.push(data.extStyle);
    } else {
      Object.keys(data.extStyle).forEach(key => {
        style.push(`${key}:${data.extStyle[key]}`);
      });
    }
  }
  if (data.rotate) {
    const REGEXP = /^-?[0-9]+(.[0-9]+)?$/;
    if (REGEXP.test(`${data.rotate}`)) {
      data.rotate += 'deg';
    }
    style.push(`transform:rotate(${data.rotate})`);
  }
  if (data.size) {
    style.push(`font-size:${addUnit(data.size)}`);
  }
  if (data.iconStyle && ['lovely', 'popular'].indexOf(data.iconStyle) !== -1) {
    style.push(`--icon-family:${handle('icon', [data.iconStyle])},${join('icon')}`);
  }
  if (data.color) {
    style.push(`color:${data.color}`);
  }
  if (data.prefix) {
    style.push(`--protected-icon-family: ${data.prefix}`);
  }
  return style.join(';');
}
function completedClasses(data) {
  let className: string | string[] = '';
  const root = join('icon-base');
  let fontFamily = '';

  className = [root, join('icon')];
  if (data.prefix) {
    fontFamily = data.prefix;
  } else {
    fontFamily = join('icon');
  }
  className.push(fontFamily);

  if (data.name) {
    className.push(fontFamily + data.hyphen + data.name);
  }

  if (data.spin) {
    className.push(`${root}-spin`);
  }
  return className.join(' ');
}

const defaultIconsUrl = `
@font-face {
  font-family: ${join('icon')}; /* Project id 2512151 */
  src: url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.woff2?t=1670393250377') format('woff2'),
    url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.woff?t=1670393250377') format('woff'),
    url('//at.alicdn.com/t/c/font_2512151_v1crhlxl0k.ttf?t=1670393250377') format('truetype');
}

@font-face {
  font-family: ${join('icon-lovely')}; /* Project id 2914379 */
  src: url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.woff2?t=1661930836565') format('woff2'),
    url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.woff?t=1661930836565') format('woff'),
    url('//at.alicdn.com/t/c/font_2914379_abca1akhv1o.ttf?t=1661930836565') format('truetype');
}

@font-face {
  font-family:${join('icon-popular')}; /* Project id 2914381 */
  src: url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.woff2?t=1661930832349') format('woff2'),
    url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.woff?t=1661930832349') format('woff'),
    url('//at.alicdn.com/t/c/font_2914381_3wmammyd5bl.ttf?t=1661930832349') format('truetype');
}
`;
@Component({
  tag: 'ti-icon',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
})
export class TiIcon {
  @Prop() extClass?: string;

  @Prop() extStyle?: string | JSXBase.HTMLAttributes<Record<string, unknown>>['style'];

  @Prop() name?: string;

  @Prop() color?: string;

  @Prop() size?: string | number;

  @Prop() spin?: boolean;

  @Prop() rotate?: string | number;

  @Prop() prefix: string;

  @Prop() hyphen?: string = '-';

  @Prop() iconStyle?: string;

  setIconStyle(prefix: string, fontFace: string) {
    if (!document.getElementById(prefix)) {
      const head = document.head || document.getElementsByTagName('head')[0];
      const style = document.createElement('style');
      style.id = prefix;
      style.innerText = fontFace;
      head.appendChild(style);
    }
  }

  componentWillLoad() {
    if (store.get('enableTitianIcon')) {
      const prefix = this.prefix || store.get('iconClassPrefix') || PREFIX;
      this.setIconStyle(prefix, defaultIconsUrl);
    }
  }

  render() {
    const { prefix, name, spin, hyphen, extStyle, rotate, size, color, iconStyle, extClass } = this;
    const className = `${completedClasses({ prefix, name, spin, hyphen })} ${extClass || ''}` || '';
    const style = completedStyles({ extStyle, rotate, size, color, iconStyle, prefix });
    return <Host class={className} style={stringToAttrStyle(style)} />;
  }
}
