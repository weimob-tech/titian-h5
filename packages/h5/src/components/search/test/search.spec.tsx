import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiSearch } from '../index';
import { TiIcon } from '../../icon';
import { TestContainer } from '../../common/test/container';
import { sleep } from '../../common/test';

describe('渲染 ti-search', () => {
  it('基本用法', async () => {
    const page = await newSpecPage({
      components: [TiSearch],
      template: () => <ti-search />,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('居中模式', async () => {
    const page = await newSpecPage({
      components: [TiSearch],
      template: () => <ti-search center />,
    });
    const dom = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-search-virtual-placeholder');
    dom?.dispatchEvent(new Event('click'));
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('输入', async () => {
    const page = await newSpecPage({
      components: [TiSearch],
      template: () => <ti-search placeholder="请输入" />,
    });
    const dom = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-search-input-host');
    dom?.dispatchEvent(new Event('focus'));
    dom?.dispatchEvent(new CustomEvent('input', { detail: '12' }));
    dom?.dispatchEvent(new CustomEvent('change', { detail: '12' }));
    dom?.dispatchEvent(new CustomEvent('confirm', { detail: '12' }));
    await sleep(1000);
    dom?.dispatchEvent(new CustomEvent('clear'));
    dom?.dispatchEvent(new Event('blur'));
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
  it('搜索按钮状态', async () => {
    const page = await newSpecPage({
      components: [TiSearch, TestContainer],
      template: () => (
        <test-container>
          <ti-search animation={false} />
          <ti-search always-show-search />
          <ti-search use-search-button={false} />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
  it('使用插槽', async () => {
    const page = await newSpecPage({
      components: [TiSearch, TiIcon, TestContainer],
      template: () => (
        <test-container>
          <ti-search>
            <div slot="prefix">
              <span>请选择</span>
              <ti-icon name="arrow-down" size={32} />
            </div>
            <ti-icon slot="right-icon" name="scan" size={32} />
          </ti-search>
          <ti-search>
            <div slot="prefix">
              <ti-icon name="category" size={42} />
            </div>
            <div slot="suffix">
              <ti-icon name="arrange" size={42} />
            </div>
          </ti-search>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
