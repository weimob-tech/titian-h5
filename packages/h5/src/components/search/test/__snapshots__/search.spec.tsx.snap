// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-search 使用插槽 1`] = `
<test-container>
  <!---->
  <ti-search>
    <mock:shadow-root>
      <div class="titian-search" part="">
        <slot name="prefix"></slot>
        <div class="titian-search-input-container" part="">
          <div style="display: contents;">
            <ti-icon class="search-icon titian-icon titian-icon-base titian-icon-search" ext-class="search-icon" style="font-size: 16px;"></ti-icon>
            <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
            <div class="titian-search-suffix-icon">
              <slot name="right-icon"></slot>
            </div>
          </div>
        </div>
        <div aria-hidden="true" class="titian-search-button-box" part="">
          <ti-button variant="text">
            搜索
          </ti-button>
        </div>
        <slot name="suffix"></slot>
      </div>
    </mock:shadow-root>
    <div slot="prefix">
      <span>
        请选择
      </span>
      <ti-icon class="titian-icon titian-icon-arrow-down titian-icon-base" style="font-size: 16px;"></ti-icon>
    </div>
    <ti-icon class="titian-icon titian-icon-base titian-icon-scan" slot="right-icon" style="font-size: 16px;"></ti-icon>
  </ti-search>
  <ti-search>
    <mock:shadow-root>
      <div class="titian-search" part="">
        <slot name="prefix"></slot>
        <div class="titian-search-input-container" part="">
          <div style="display: contents;">
            <ti-icon class="search-icon titian-icon titian-icon-base titian-icon-search" ext-class="search-icon" style="font-size: 16px;"></ti-icon>
            <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
            <div class="titian-search-suffix-icon">
              <slot name="right-icon"></slot>
            </div>
          </div>
        </div>
        <div aria-hidden="true" class="titian-search-button-box" part="">
          <ti-button variant="text">
            搜索
          </ti-button>
        </div>
        <slot name="suffix"></slot>
      </div>
    </mock:shadow-root>
    <div slot="prefix">
      <ti-icon class="titian-icon titian-icon-base titian-icon-category" style="font-size: 21px;"></ti-icon>
    </div>
    <div slot="suffix">
      <ti-icon class="titian-icon titian-icon-arrange titian-icon-base" style="font-size: 21px;"></ti-icon>
    </div>
  </ti-search>
</test-container>
`;

exports[`渲染 ti-search 基本用法 1`] = `
<ti-search>
  <mock:shadow-root>
    <div class="titian-search" part="">
      <slot name="prefix"></slot>
      <div class="titian-search-input-container" part="">
        <div style="display: contents;">
          <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
          <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
          <div class="titian-search-suffix-icon">
            <slot name="right-icon"></slot>
          </div>
        </div>
      </div>
      <div aria-hidden="true" class="titian-search-button-box" part="">
        <ti-button variant="text">
          搜索
        </ti-button>
      </div>
      <slot name="suffix"></slot>
    </div>
  </mock:shadow-root>
</ti-search>
`;

exports[`渲染 ti-search 居中模式 1`] = `
<ti-search>
  <mock:shadow-root>
    <div class="titian-search" part="">
      <div class="titian-search-input-container" part="">
        <div style="display: contents;">
          <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
          <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
        </div>
      </div>
      <div aria-hidden="true" class="titian-search-button-box titian-search-button-box-focus" part="">
        <ti-button variant="text">
          搜索
        </ti-button>
      </div>
    </div>
  </mock:shadow-root>
</ti-search>
`;

exports[`渲染 ti-search 搜索按钮状态 1`] = `
<test-container>
  <!---->
  <ti-search>
    <mock:shadow-root>
      <div class="titian-search" part="">
        <slot name="prefix"></slot>
        <div class="titian-search-input-container" part="">
          <div style="display: contents;">
            <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
            <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
            <div class="titian-search-suffix-icon">
              <slot name="right-icon"></slot>
            </div>
          </div>
        </div>
        <div aria-hidden="true" class="titian-search-button-box titian-search-button-box-pure" part="">
          <ti-button variant="text">
            搜索
          </ti-button>
        </div>
        <slot name="suffix"></slot>
      </div>
    </mock:shadow-root>
  </ti-search>
  <ti-search always-show-search="">
    <mock:shadow-root>
      <div class="titian-search" part="">
        <slot name="prefix"></slot>
        <div class="titian-search-input-container" part="">
          <div style="display: contents;">
            <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
            <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
            <div class="titian-search-suffix-icon">
              <slot name="right-icon"></slot>
            </div>
          </div>
        </div>
        <div aria-hidden="true" class="titian-search-button-box titian-search-button-box-focus" part="">
          <ti-button variant="text">
            搜索
          </ti-button>
        </div>
        <slot name="suffix"></slot>
      </div>
    </mock:shadow-root>
  </ti-search>
  <ti-search>
    <mock:shadow-root>
      <div class="titian-search" part="">
        <slot name="prefix"></slot>
        <div class="titian-search-input-container" part="">
          <div style="display: contents;">
            <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
            <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class=""></ti-input>
            <div class="titian-search-suffix-icon">
              <slot name="right-icon"></slot>
            </div>
          </div>
        </div>
        <div aria-hidden="true" class="titian-search-button-box" part="">
          <ti-button variant="text">
            搜索
          </ti-button>
        </div>
        <slot name="suffix"></slot>
      </div>
    </mock:shadow-root>
  </ti-search>
</test-container>
`;

exports[`渲染 ti-search 输入 1`] = `
<ti-search>
  <mock:shadow-root>
    <div class="titian-search" part="">
      <slot name="prefix"></slot>
      <div class="titian-search-input-container" part="">
        <div style="display: contents;">
          <ti-icon ext-class="search-icon" name="search" size="32"></ti-icon>
          <ti-input class="titian-search-input-host" clearable="" confirm-type="search" exportparts="" ext-class="search-input" ext-css="" input-class="" placeholder="请输入"></ti-input>
          <div class="titian-search-suffix-icon">
            <slot name="right-icon"></slot>
          </div>
        </div>
      </div>
      <div aria-hidden="true" class="titian-search-button-box" part="">
        <ti-button variant="text">
          搜索
        </ti-button>
      </div>
      <slot name="suffix"></slot>
    </div>
  </mock:shadow-root>
</ti-search>
`;
